crates/stack/src/socket.rs

```diff
@@ -214,6 +217,24 @@ impl<T: Into<IpAddress>> From<(T, u16)> for SocketEndpoint {
     }
 }
 
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum RecvError {
```

Comment:
smoltcp 0.8 had a common `RecvError` type.
To limit necessary changes I created similar error.